featuretools/utils/recommend_primitives.py

```diff
@@ -0,0 +1,261 @@
+from typing import List
+
+import pandas as pd
+
+from featuretools.computational_backends import calculate_feature_matrix
+from featuretools.entityset import EntitySet
+from featuretools.primitives.utils import get_transform_primitives
+from featuretools.synthesis import dfs, get_valid_primitives
+
+ORDERED_PRIMITIVES = [
+    "absolute_diff",
+    "cum_sum",
+    "cum_count",
+    "cum_mean",
+    "cum_max",
+    "cum_min",
+    "cumulative_time_since_last_false",  # time_index
+    "cumulative_time_since_last_true",  # time_index
+    "diff",
+    "diff_datetime",
+    "exponential_weighted_average",
+    "exponential_weighted_std",
+    "exponential_weighted_variance",
+    "greater_than_previous",
+    "is_first_occurrence",
+    "is_last_occurrence",
+    "is_max_so_far",
+    "is_min_so_far",
+    "lag",  # time_index
+    "less_than_previous",
+    "percent_change",
+    "same_as_previous",
+    "time_since_previous",  # time_index
+]
+
+
+DEPRECATED_PRIMITIVES = [
+    # // as of premium - primitives 0.14.0 and feature - tools 0.26.1
+    "multiply_boolean",  # functionality duplicated by 'and' primitive
+    "numeric_lag",  # deperecated and replaced with `lag`
+]
+
+REQUIRED_INPUT_PRIMITIVES = [
+    # // as of premium - primitives 0.14.0 and feature - tools 0.26.1
+    "count_string",
+    "distance_to_holiday",
+    "is_in_geobox",
+    "score_percentile",
+    "subtract_numeric_scalar",
+    "scalar_subtract_numeric_feature",
+    "not_equal_scalar",
+    "multiply_numeric_scalar",
+    "modulo_numeric_scalar",
+    "divide_numeric_scalar",
+    "add_numeric_scalar",
+    "equal_scalar",
+    "greater_than_equal_to_scalar",
+    "less_than_equal_to_scalar",
+    "divide_by_feature",
+    "greater_than_scalar",
+    "less_than_scalar",
+    "modulo_by_feature",
+    "time_since",
+    "savgol_filter",
+    "isin",
+    "numeric_bin",
+]
+
+OTHER_PRIMITIVES_TO_EXCLUDE = [
+    "absolute",
+    "negate",
+    "not",
+    "and",
+    "is_zero",
+    "is_null",
+    "greater_than_equal_to",
+    "equal",
+    "greater_than",
+    "is_whole_number",
+    "not_equal",
+    "multiply_numeric",
+    "add_numeric",
+    "less_than_equal_to",
+    "divide_numeric",
+    "less_than",
+    "subtract_numeric",
+    "modulo_numeric",
+    "cosine",
+    "tangent",
+    "sine",
+]
+
+DEFAULT_EXCLUDED_PRIMITIVES = (
+    REQUIRED_INPUT_PRIMITIVES
+    + DEPRECATED_PRIMITIVES
+    + ORDERED_PRIMITIVES
+    + OTHER_PRIMITIVES_TO_EXCLUDE
+)
+
+TIME_SERIES_PRIMITIVES = [
```

Comment:
Just curious, is the plan to include Expanding primitives here?

Reply:
I think so I wanted to get the group's perspective but I'm going to assume the answer is yes and make this change.

Reply:
I think we need to include all the time series only primitives, because if we don't they won't get added to the list of excluded primitives for non-time series problems, correct? Also we have a RollingOutlierCount primitive that needs to be included.

We really need to have a better way to identify these primitives longer term. I'm afraid manually maintaining this list will lead to problems long term (but that is out of scope for this PR).